presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -674,6 +774,209 @@ public String toString()
                 .toString();
     }
 
+    private void rollbackInsertChanges(HiveInsertTableHandle handle, Map<String, List<String>> filesWritten)
+    {
+        Path tableLocation = new Path(handle.getTargetPath());
+        try {
+            for (String partition : filesWritten.keySet()) {
+                Path partitionPath;
+                if (handle.isOutputTablePartitioned()) {
+                    partitionPath = new Path(tableLocation, partition);
+                }
+                else {
+                    partitionPath = tableLocation;
+                }
+                for (String file : filesWritten.get(partition)) {
+                    delete(new Path(partitionPath, file), false);
+                }
+            }
+        }
+        catch (IOException e) {
+            log.info(String.format("Files with prefix %s need to be deleted manually from location %s, Rollback of changes made during insert failed with %s",
```

Comment:
This method is called when commitInsert hits an exception. We want commitInsert's exception to be bubbled up and not this methods, hence swallowing it.